### PR TITLE
core: proxy: Log errors at register and config

### PR DESCRIPTION
**Pull request by jcorde**
Log errors when registering or configuring a proxy.

Signed-off-by: Joao Cordeiro <jvcc@cesar.org.br>

### CODE REVIEW

File: core/src/proxy.c [hunk 1]
```diff
@@ -100,6 +100,12 @@ void proxy_stop(void)
 
 }
 
+static void *register_fail(u8_t id)
```
Comment (ramonhpr) [inline core/src/proxy.c]: This function is not necessary

File: core/src/proxy.c [hunk 2]
```diff
@@ -108,16 +114,23 @@ struct knot_proxy *knot_proxy_register(u8_t id, const char *name,
 	struct knot_proxy *proxy;
 
 	/* Out of index? */
-	if (id >= CONFIG_KNOT_THING_DATA_MAX)
-		return NULL;
+	if (id >= CONFIG_KNOT_THING_DATA_MAX) {
+		LOG_ERR("ID %d: id >= CONFIG_KNOT_THING_DATA_MAX (%d)",
+			id, CONFIG_KNOT_THING_DATA_MAX);
+		return register_fail(id);
```
Comment (ramonhpr) [inline core/src/proxy.c]: `return NULL;`
You are already logging the reason why the register fails there is no need to log where was the fail. I think that with zephyr logger you can enable the callback trace if you really need to know what function raise this logging.

File: core/src/proxy.c [hunk 3]
```diff
@@ -108,16 +114,23 @@ struct knot_proxy *knot_proxy_register(u8_t id, const char *name,
...
+	}
 
 	/* Assigned already? */
-	if (proxy_pool[id].id != 0xff)
-		return NULL;
+	if (proxy_pool[id].id != 0xff) {
+		LOG_ERR("ID %d: Id already assigned", id);
+		return register_fail(id);
```
Comment (ramonhpr) [inline core/src/proxy.c]: `return NULL;`

File: core/src/proxy.c [hunk 4]
```diff
@@ -108,16 +114,23 @@ struct knot_proxy *knot_proxy_register(u8_t id, const char *name,
...
+	}
 
 	/* Basic field validation */
-	if (knot_schema_is_valid(type_id, value_type, unit) != 0 || !name)
-		return NULL;
+	if (knot_schema_is_valid(type_id, value_type, unit) != 0 || !name) {
+		LOG_ERR("ID %d: Invalid schema", id);
+		return register_fail(id);
```
Comment (ramonhpr) [inline core/src/proxy.c]: `return NULL;`

File: core/src/proxy.c [hunk 5]
```diff
@@ -159,13 +178,18 @@ bool knot_proxy_set_config(u8_t id, ...)
 	lower_limit.val_i = 0;
 	upper_limit.val_i = 0;
 
-	if (id >= CONFIG_KNOT_THING_DATA_MAX)
-		return false;
+	if (id >= CONFIG_KNOT_THING_DATA_MAX) {
+		LOG_ERR("ID %d: id >= CONFIG_KNOT_THING_DATA_MAX (%d)",
+			id, CONFIG_KNOT_THING_DATA_MAX);
+		return config_fail(id);
```
Comment (ramonhpr) [inline core/src/proxy.c]: `return false;`

File: core/src/proxy.c [hunk 6]
```diff
@@ -159,13 +178,18 @@ bool knot_proxy_set_config(u8_t id, ...)
...
+	}
 
 	proxy = &proxy_pool[id];
 
-	if (proxy->id != id)
-		return false;
+	if (proxy->id != id) {
+		LOG_ERR("ID %d: Proxy not found!", id);
+		return config_fail(id);
```
Comment (ramonhpr) [inline core/src/proxy.c]: `return false;`

File: core/src/proxy.c [hunk 7]
```diff
@@ -201,15 +225,18 @@ bool knot_proxy_set_config(u8_t id, ...)
 			break;
 		default:
 			va_end(event_args);
-			return false;
+			LOG_ERR("ID %d: Invalid config flags", id);
+			return config_fail(id);
```
Comment (ramonhpr) [inline core/src/proxy.c]: `return false;`

File: core/src/proxy.c [hunk 8]
```diff
@@ -201,15 +225,18 @@ bool knot_proxy_set_config(u8_t id, ...)
...
 
 	} while(event);
 	va_end(event_args);
 
 	if (knot_config_is_valid(event_flags, proxy->schema.value_type,
-		 timeout_sec, &lower_limit, &upper_limit) != 0)
-		return false;
+				 timeout_sec, &lower_limit, &upper_limit) != 0) {
+		LOG_ERR("ID %d: Invalid config values", id);
+		return config_fail(id);
```
Comment (ramonhpr) [inline core/src/proxy.c]: `return false;`

File: core/src/proxy.c [hunk 9]
```diff
@@ -145,6 +158,12 @@ struct knot_proxy *knot_proxy_register(u8_t id, const char *name,
 	return proxy;
 }
 
+bool config_fail(u8_t id)
```
Comment (ramonhpr) [inline core/src/proxy.c]: this function is not necessary

**Review by ramonhpr — CHANGES_REQUESTED** [review]

File: core/src/proxy.c (hunk 2 above)
Comment (jcorde) [reply core/src/proxy.c]: Zephyr logs the function that failed it. But yet I think it is not explicit enough

**Pull request merged**